File: docs/usermanual/source/processing/scripting/bufferSplit.py
from geoserver.wps import process
from geoscript.process import Process
from geoscript.geom import *

buffer = Process.lookup('JTS:buffer')
split = Process.lookup('JTS:splitPolygon')

@process(
  title = 'Buffer and Split',
  description = 'Buffers a geometry and splits the resulting polygon.',
  inputs = {
    'geom': (Geometry, 'The geometry to buffer'),
    'dist': (float, 'The distance by which to buffer the target geometry'),
    'line': (LineString, 'The line used for splitting the buffered geometry')
  },
  outputs = {
    'result': (Geometry, 'The buffered and split geometry')
  }
)
def run(geom, dist, line):

  buffered = buffer.run(geom=geom, distance=dist)['result']
  return split.run(polygon=buffered, line=line)['result']

File: docs/usermanual/source/processing/scripting/bufferSplit.js
var Process = require("geoscript/process").Process;

var buffer = Process.get("JTS:buffer");
var split = Process.get("JTS:splitPolygon");

exports.process = new Process({
  title: "Buffer and Split",
  description: "Buffers a geometry and splits the resulting polygon",
  inputs: {
    geom: {
      type: "Geometry",
      title:"Target Geometry",
      description: "The geometry to buffer"
    },
    dist: {
      type: "Number",
      title: "Buffer Distance",
      description: "The distance by which to buffer the target geometry"
    },
    line: {
      type: "LineString",
      title: "Splitter Line",
      description: "The line used for splitting the buffered geometry"
    }
  },
  outputs: {
    result: {
      type: "Geometry",
      title: "Result",
      description: "The buffered and split geometry"
    }
  },
  run: function(inputs) {
    var buffered = buffer.run({
      geom: inputs.geom, distance: inputs.dist
    });
        
    return split.run({polygon: buffered.result, line: inputs.line});
  }
});
